### DISCUSSION TITLE
SVCB and HTTPS records question

### DISCUSSION THREAD

**Discussion by ammnt** [AdguardTeam/AdGuardHome#2976] [Q&A]
Hello,

it's just a question... I'd appreciate any advice. How to properly generate SVCB and HTTPS records if the my AGH DNS server has **only one IPv4 address** and **only DoH protocol** will be used (no DOQ)?🤔

Do I need them if the server is **only one** and with **one IPv4 record**? Will there be a conflict here if A record is always overwritten via **/etc/hosts**?

Thank you.
Cheers✌️

**Comment by ammnt** [comment]
No one know? No one care?🥺

**Comment by ameshkov** [comment]
The question is rather confusing. Why does it matter how many IPs or protocols your server has?

Regarding HTTPS records, check out the doc on $dnsrewrite:
https://github.com/AdguardTeam/AdGuardHome/wiki/Hosts-Blocklists#dnsrewrite

**Reply by ammnt** [reply]
@ameshkov, thank you for your answer! But I actually mean not dnsrewrite rules, but how to configure HTTPS and SVCB records for my DNS server himself😁

**Reply by ameshkov** [reply]
HTTPS records may contain different instructions. What exactly are you trying to configure and achieve with it?

**Reply by ammnt** [reply]
@ameshkov, i just want to speeding up HTTPS negotiation with DNS SVCB and HTTPS records (for DoH and maybe DoQ protocols in the future). I found few samples [here](https://blog.cloudflare.com/speeding-up-https-and-http-3-negotiation-with-dns/):
```
example.com 3600 IN HTTPS 1 . alpn=”h3,h2” ipv4hint=”192.0.2.1” ipv6hint=”2001:db8::1”
example.com 3600 IN HTTPS 1 example.net alpn=”h3,h2”
example.com 3600 IN HTTPS 2 example.org alpn=”h2”
```

I'm  not sure how to propely configure these record for my DNS-server, for my stub-domain for DoH/DoQ. Can you try help me please to figure out?🙃

Thank you.

**Reply by ameshkov** [reply]
Note, that HTTPS records won't speed up anything because your browser will anyway try to initiate a QUIC connection alongside normal TCP connection.

There're examples of how to add HTTPS records (including ipv4hint) in the $dnsrewrite doc.
